akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -3726,4 +3726,64 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
   def monitor: ReprMat[Out, (Mat, FlowMonitor[Out])] =
     monitorMat(Keep.both)
 
+  /**
+   * This is a more scalable and general case of [[groupedWeightedWithin]]
+   * which groups a stream into vectors based on custom weight and time.
+   * The problem with that solution is each grouped vector must fit into memory before emitting to the next stage.
+   * That won't work for the use case of writing large files.
+   * The desirable behavior is to write data as they come as opposed to accumulate everything until the group slicing condition is met.
+   * In this case, the output channel needs to be closed if there is no data arriving within certain time to avoid connection timeout.
+   * This custom flow uses custom aggregator to support such use cases.
+   * Upstream inputs are continuously aggregated as they arrive.
+   * The currently aggregated elements can be terminated and emitted based on custom conditions
+   *
+   * @param allocate    allocate the initial data structure for aggregated elements
+   * @param aggregate   update the aggregated elements, return true if ready to emit after update
+   * @param emitOnTimer decide whether the currently aggregated elements can be emitted on each timer event
```

Comment:
Add more details about that the timer function is invoked with the current aggregate on every interval to decide if it should be emitted here